# MID project DA

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re

### 1. data

##### 1. 인구 이동 데이터

In [15]:
data_trans = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\시군구별_이동자수.csv'
df_trans = pd.read_csv(data_trans)
df_trans.head()

행정구역(시군구)별  2023.01 2023.01.1 2023.01.2       2023.01.3          2023.01.4  \
0  행정구역(시군구)별  총전입 (명)   총전출 (명)   순이동 (명)  시도내이동-시군구내 (명)  시도내이동-시군구간 전입 (명)   
1          전국   506100    506100         0          164141             145923   
2       서울특별시    91225     94248     -3023           19465              35007   
3       부산광역시    31180     31800      -620            8601              12723   
4       대구광역시    22316     22557      -241            5853               8343   

           2023.01.5  2023.01.6  2023.01.7  2023.02  ...     2021.6  \
0  시도내이동-시군구간 전출 (명)  시도간전입 (명)  시도간전출 (명)  총전입 (명)  ...  시도간전입 (명)   
1             145923     196036     196036   622411  ...    2447234   
2              35007      36753      39776   126733  ...     461123   
3              12723       9856      10476    40851  ...     116166   
4               8343       8120       8361    26929  ...      83076   

      2021.7     2022   2022.1   2022.2          2022.3             2022.4  \
0  시도간전출 (명)  총전입 (명)  총전출 (명)  순이동 (명)  시도내이동-시군구내 (명)  시도내이동-시군구간 전입 (명)   
1    2447234  6152155  6152155        0         2173548            1805002   
2     567366  1201527  1236867   -35340          302805             468146   
3     135069   383848   397410   -13562          109335             169621   
4     107395   265354   276873   -11519           81016             102771   

              2022.5     2022.6     2022.7  
0  시도내이동-시군구간 전출 (명)  시도간전입 (명)  시도간전출 (명)  
1            1805002    2173605    2173605  
2             468146     430576     465916  
3             169621     104892     118454  
4             102771      81567      93086  

[5 rows x 496 columns]

In [4]:
data_trans_age = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\Sigungu_age_trans.csv'
df_trans_age = pd.read_csv(data_trans_age, encoding='cp949')
df_trans_age.drop('Unnamed: 66', axis=1, inplace=True)
df_trans_age.head()


행정구역(시군구)별 연령별                항목 단위  2023.01 월  2023.02 월  2023.03 월  \
0         전국   계            총전입[명]  명     506100     622411     588958   
1         전국   계            총전출[명]  명     506100     622411     588958   
2         전국   계            순이동[명]  명          0          0          0   
3         전국   계     시도내이동-시군구내[명]  명     164141     187696     195928   
4         전국   계  시도내이동-시군구간 전입[명]  명     145923     183174     171026   

   2023.04 월  2023.05 월  2023.06 월  ...   2013 년   2014 년   2015 년   2016 년  \
0     455474     497285     473480  ...  7411784  7629098  7755286  7378430   
1     455474     497285     473480  ...  7411784  7629098  7755286  7378430   
2          0          0          0  ...        0        0        0        0   
3     161731     180983     173203  ...  2864028  2957867  3014020  2840295   
4     136415     148019     140977  ...  2124327  2163435  2189842  2084793   

    2017 년   2018 년   2019 년   2020 년   2021 년   2022 년  
0  7154226  7297099  7104398  7735491  7213422  6152155  
1  7154226  7297099  7104398  7735491  7213422  6152155  
2        0        0        0        0        0        0  
3  2701400  2776681  2659826  2926406  2649108  2173548  
4  2041896  2091234  2059624  2274971  2117080  1805002  

[5 rows x 66 columns]

### 2. DA

##### 1. 인구 이동 추이

In [16]:
df_trans.drop(0, inplace=True)
df_trans.head()

행정구역(시군구)별 2023.01 2023.01.1 2023.01.2 2023.01.3 2023.01.4 2023.01.5  \
1         전국  506100    506100         0    164141    145923    145923   
2      서울특별시   91225     94248     -3023     19465     35007     35007   
3      부산광역시   31180     31800      -620      8601     12723     12723   
4      대구광역시   22316     22557      -241      5853      8343      8343   
5      인천광역시   30920     28525      2395      8797      8747      8747   

  2023.01.6 2023.01.7 2023.02  ...   2021.6   2021.7     2022   2022.1  \
1    196036    196036  622411  ...  2447234  2447234  6152155  6152155   
2     36753     39776  126733  ...   461123   567366  1201527  1236867   
3      9856     10476   40851  ...   116166   135069   383848   397410   
4      8120      8361   26929  ...    83076   107395   265354   276873   
5     13376     10981   38362  ...   155824   144401   395140   367039   

   2022.2   2022.3   2022.4   2022.5   2022.6   2022.7  
1       0  2173548  1805002  1805002  2173605  2173605  
2  -35340   302805   468146   468146   430576   465916  
3  -13562   109335   169621   169621   104892   118454  
4  -11519    81016   102771   102771    81567    93086  
5   28101   123107   116812   116812   155221   127120  

[5 rows x 496 columns]

In [18]:
# 2023년 월별 분리된 데이터 통합: 2023 1월 ~ 9월 데이터

df_trans["2023.2"] = df_trans["2023.01.2"].apply(lambda x: int(x))
for i in range(2, 10):
    df_trans["2023.2"] += df_trans[f"2023.0{i}.2"].apply(lambda x: int(x))
df_trans["2023.2"]

1         0
2    -16467
3     -8147
4     -4834
5     22371
6     -6484
7      -891
8     -6109
9      1326
10    35049
11     -882
12     2986
13    10603
14    -3710
15    -2140
16    -7027
17   -14556
18    -1088
Name: 2023.2, dtype: int64

In [23]:
# 년도별 순이동 인구 데이터 선별

column_li = df_trans.columns

# ".2"로 끝나면서 "####.##.2" 형식이 아닌 항목 찾기
result_list = [column for column in column_li if re.search(r'\.2$', column) and not re.search(r'\d{4}\.\d{2}\.2$', column)]

# DataFrame
result_list.insert(0, "행정구역(시군구)별")
df_trans_diff = df_trans[result_list]
df_trans_diff


행정구역(시군구)별  1970.2  1971.2  1972.2  1973.2  1974.2   1975.2  1976.2  \
1          전국       0       0       0       0       0        0       0   
2       서울특별시  293828  252689  138774  184590  230732   459569  260890   
3       부산광역시   72518   61430   42842   67400  126971   111534   95499   
4       대구광역시       -       -       -       -       -        -       -   
5       인천광역시       -       -       -       -       -        -       -   
6       광주광역시       -       -       -       -       -        -       -   
7       대전광역시       -       -       -       -       -        -       -   
8       울산광역시       -       -       -       -       -        -       -   
9     세종특별자치시       -       -       -       -       -        -       -   
10        경기도   98525  121277   35100   87798   94597   181560  124094   
11    강원특별자치도  -36222  -34994  -30734  -35802  -49300   -60210  -41882   
12       충청북도  -47875  -44692  -24957  -33921  -40750   -78248  -49003   
13       충청남도  -47262  -50341  -22371  -38158  -51328   -80524  -51981   
14       전라북도  -91270  -74495  -37017  -53076  -71645  -137347  -86763   
15       전라남도  -86372  -85649  -41002  -59451  -83974  -133291  -71088   
16       경상북도  -62499  -57379  -13590  -25200  -27208   -83254  -58531   
17       경상남도  -87276  -83934  -47696  -59260  -55326  -116412  -95125   
18    제주특별자치도   -3154   -3912     651   -2917   -5577    -9901   -7042   

    1977.2  1978.2  ...  2014.2   2015.2   2016.2  2017.2   2018.2  2019.2  \
1        0       0  ...       0        0        0       0        0       0   
2   215396  278528  ...  -87831  -137256  -140257  -98486  -110230  -49588   
3    88997  112359  ...  -15092   -13560   -21392  -28398   -26759  -23354   
4        -       -  ...  -15526   -12940    -9260  -11936   -14242  -23673   
5        -       -  ...    9324     9538     5777   -1670      -67   -2337   
6        -       -  ...   -2977    -9272    -7898   -8118    -6216   -3875   
7        -       -  ...   -8838   -20616   -10631  -16175   -14753  -16342   
8        -       -  ...    2786      -80    -7622  -11917   -12654  -10172   
9        -       -  ...   33456    53044    29816   34690    31433   23724   
10  126455  168318  ...   57396    94768   133617  116162   170094  134666   
11  -45713  -50777  ...    2085     4773     1892    2035    -3854    1661   
12  -56369  -84617  ...    2367     1438     5011    2262     5149    2565   
13  -43393  -37483  ...    9810    10272    15318   19401    10084    -794   
14  -81468  -94771  ...   -2563    -2830    -4419   -7206   -13773  -12748   
15  -76770  -90449  ...    -596     3936    -3205   -3047    -8030   -8788   
16  -61277  -62378  ...    -988     -577    -3151   -5581    -9225   -4571   
17  -52227  -45105  ...    6075     5105     1772    3979    -5810   -9310   
18   -8395   -9636  ...   11112    14257    14632   14005     8853    2936   

    2020.2   2021.2  2022.2 2023.2  
1        0        0       0      0  
2   -64850  -106243  -35340 -16467  
3   -14347   -18903  -13562  -8147  
4   -16835   -24319  -11519  -4834  
5   -15748    11423   28101  22371  
6    -6083    -5883   -7642  -6484  
7   -11097    -8931   -2996   -891  
8   -13584   -13674   -9536  -6109  
9    13025    14085   10128   1326  
10  168373   150517   43882  35049  
11    5457     6681    7494   -882  
12    3454     3462    5201   2986  
13     741     8522   14314  10603  
14   -8494    -5801   -5115  -3710  
15   -9754    -4487    -345  -2140  
16  -16978     3337   -7666  -7027  
17  -16658   -13703  -18547 -14556  
18    3378     3917    3148  -1088  

[18 rows x 55 columns]

In [22]:
[column for column in column_li]

['행정구역(시군구)별',
 '2023.01',
 '2023.01.1',
 '2023.01.2',
 '2023.01.3',
 '2023.01.4',
 '2023.01.5',
 '2023.01.6',
 '2023.01.7',
 '2023.02',
 '2023.02.1',
 '2023.02.2',
 '2023.02.3',
 '2023.02.4',
 '2023.02.5',
 '2023.02.6',
 '2023.02.7',
 '2023.03',
 '2023.03.1',
 '2023.03.2',
 '2023.03.3',
 '2023.03.4',
 '2023.03.5',
 '2023.03.6',
 '2023.03.7',
 '2023.04',
 '2023.04.1',
 '2023.04.2',
 '2023.04.3',
 '2023.04.4',
 '2023.04.5',
 '2023.04.6',
 '2023.04.7',
 '2023.05',
 '2023.05.1',
 '2023.05.2',
 '2023.05.3',
 '2023.05.4',
 '2023.05.5',
 '2023.05.6',
 '2023.05.7',
 '2023.06',
 '2023.06.1',
 '2023.06.2',
 '2023.06.3',
 '2023.06.4',
 '2023.06.5',
 '2023.06.6',
 '2023.06.7',
 '2023.07',
 '2023.07.1',
 '2023.07.2',
 '2023.07.3',
 '2023.07.4',
 '2023.07.5',
 '2023.07.6',
 '2023.07.7',
 '2023.08',
 '2023.08.1',
 '2023.08.2',
 '2023.08.3',
 '2023.08.4',
 '2023.08.5',
 '2023.08.6',
 '2023.08.7',
 '2023.09',
 '2023.09.1',
 '2023.09.2',
 '2023.09.3',
 '2023.09.4',
 '2023.09.5',
 '2023.09.6',
 '2023.09

# MID Project Employ DA

### 1. data

##### 1. 행정구역(시도)/산업별 취업자 (단위: 천명)
- 시도별 산업 종사자 격차 파악 (산업별 현황 비율)
- 년도별, 시도별 취업자 수 변화량과 추이
- 산업별 가중치?

In [2]:
data1 = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\행정구역_시도__산업별_취업자.csv'
df_emp = pd.read_csv(data1)
df_emp.head()

시도별                   산업별  2023.01  2023.02  2023.03  2023.04  2023.05  \
0   계                     계    27363    27714    28223    28432    28835   
1   계  A 농업, 임업 및 어업(01~03)     1199     1231     1438     1557     1654   
2   계            * 광공업(B,C)     4438     4429     4470     4428     4474   
3   계           B 광업(05~08)        7        7        8        7        9   
4   계          C 제조업(10~34)     4432     4423     4463     4421     4464   

   2023.06  2023.07  2023.08  ...   2013   2014   2015   2016   2017   2018  \
0    28812    28686    28678  ...  25299  25897  26178  26409  26725  26822   
1     1663     1622     1644  ...   1513   1446   1337   1273   1279   1340   
2     4498     4493     4468  ...   4322   4471   4618   4603   4589   4529   
3       10        7        8  ...     16     13     14     19     23     19   
4     4488     4485     4460  ...   4307   4459   4604   4584   4566   4510   

    2019   2020   2021   2022  
0  27123  26904  27273  28089  
1   1395   1445   1458   1526  
2   4444   4389   4380   4512  
3     15     13     12      9  
4   4429   4376   4368   4503  

[5 rows x 21 columns]

- 행정구역(시도)/산업별 취업자 연령별

In [20]:
data2 = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\산업별 취업자_성_연령별.csv'
df_emp_age = pd.read_csv(data2, encoding='cp949')
df_emp_age.drop('Unnamed: 25', axis=1, inplace=True)
df_emp_age.head()

성별 행정구역별 연령별               항목  단위    2013. 1/2    2013. 2/2    2014. 1/2  \
0  전체    전국   계               합계 NaN  25186.58626  25657.67830  25834.60370   
1  전체    전국   계             농림어업 NaN   1585.58672   1742.63917   1556.08159   
2  전체    전국   계               광업 NaN     17.51169     14.84445     14.07833   
3  전체    전국   계              제조업 NaN   4255.06091   4279.32812   4360.50622   
4  전체    전국   계  전기,가스,증기 및 수도사업 NaN     89.55611     88.43435     83.69547   

     2014. 2/2    2015. 1/2  ...    2018. 1/2    2018. 2/2    2019. 1/2  \
0  26099.03937  25991.48465  ...  26711.29366  26924.74284  26870.72813   
1   1602.28053   1421.76187  ...   1393.56601   1492.53897   1406.33668   
2      8.68305     13.35277  ...     22.30169     14.89440     15.10557   
3   4419.64827   4507.11464  ...   4417.74344   4454.14493   4365.90571   
4     86.08151     93.61092  ...     91.67292     84.63680     85.23512   

     2019. 2/2    2020. 1/2    2020. 2/2    2021. 1/2    2021. 2/2  \
0  27346.83918  26397.32881  26916.90112  27213.59232  27740.74699   
1   1551.32599   1479.33688   1573.95216   1476.33225   1593.99323   
2     14.85899     14.79954      10.5006     12.76848     10.66067   
3   4381.63380   4322.13703   4283.33287   4386.11068   4323.55863   
4     88.89853     83.41409      93.8141     64.53602     76.66299   

     2022. 1/2    2022. 2/2  
0  28078.30968  28417.86366  
1   1544.66108   1640.60640  
2     13.10791      6.12520  
3   4517.98609   4524.94367  
4     76.05670     84.03880  

[5 rows x 25 columns]

##### 2. 전국 시군구 단위 소득 구간대별 주민 비율
- 소득 2천만원 주민 비율 ~ 소득 7천만원 이상 주민 비율
- 2023.09

In [5]:
data3 = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\시군구단위소득구간대.csv'
df_inc = pd.read_csv(data3)
df_inc.head()

BASE_YM    SIGNGU_NM  SIGNGU_CD  INCOME_TWO_TMW_INHBT_RATE  \
0   202309  강원특별자치도 강릉시      51150                       0.15   
1   202309  강원특별자치도 고성군      51820                       0.18   
2   202309  강원특별자치도 동해시      51170                       0.14   
3   202309  강원특별자치도 삼척시      51230                       0.16   
4   202309  강원특별자치도 속초시      51210                       0.14   

   INCOME_THREE_TMW_INHBT_RATE  INCOME_FOUR_TMW_INHBT_RATE  \
0                         0.37                        0.25   
1                         0.42                        0.23   
2                         0.36                        0.26   
3                         0.38                        0.24   
4                         0.37                        0.27   

   INCOME_FIVE_TMW_INHBT_RATE  INCOME_SIX_TMW_INHBT_RATE  \
0                        0.11                       0.05   
1                        0.08                       0.04   
2                        0.11                       0.06   
3                        0.10                       0.05   
4                        0.11                       0.05   

   INCOME_SEVEN_TMW_INHBT_RATE  INCOME_SEVEN_TMW_ABOVE_INHBT_RATE  
0                         0.03                               0.04  
1                         0.02                               0.03  
2                         0.03                               0.05  
3                         0.03                               0.04  
4                         0.03                               0.04

- 평균 임금 데이터

In [11]:
data4 = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\시군구평균임금.csv'
df_inc_mean = pd.read_csv(data4)
df_inc_mean.head()

BASE_YM    SIGNGU_NM  SIGNGU_CD  AVRG_INCOME_PRICE
0   202309  강원특별자치도 강릉시      51150              34650
1   202309  강원특별자치도 고성군      51820              31933
2   202309  강원특별자치도 동해시      51170              35029
3   202309  강원특별자치도 삼척시      51230              33474
4   202309  강원특별자치도 속초시      51210              34371

##### 3. 임금 근로 일자리 동향 행정

In [23]:
data5 = r'C:\Users\jjjun_ii\Documents\GitHub\LIKELION_AI\Project\MID_PROJECT\data\임금근로일자리동향.csv'
df_worksup = pd.read_csv(data5, encoding='cp949')
df_worksup.drop('Unnamed: 26', axis=1, inplace=True)
df_worksup.head()

연령별 산업대분류별 일자리형태별             항목  단위  2018.1/4  2018.2/4  2018.3/4  \
0  총 계    총 계    총 계  임금근로 일자리수[만개]  만개    1774.5    1822.1    1810.4   
1  총 계    총 계    총 계     산업별 구성비[%]   %     100.0     100.0     100.0   
2  총 계    총 계    총 계     형태별 구성비[%]   %     100.0     100.0     100.0   
3  총 계    총 계     지속  임금근로 일자리수[만개]  만개    1203.7    1226.6    1225.1   
4  총 계    총 계     지속     산업별 구성비[%]   %     100.0     100.0     100.0   

   2018.4/4  2019.1/4  ...  2020.4/4  2021.1/4  2021.2/4  2021.3/4  2021.4/4  \
0    1849.4    1824.8  ...    1958.9    1899.7    1957.7    1959.9    1996.5   
1     100.0     100.0  ...     100.0     100.0     100.0     100.0     100.0   
2     100.0     100.0  ...     100.0     100.0     100.0     100.0     100.0   
3    1240.8    1235.4  ...    1357.3    1350.0    1362.6    1363.9    1380.6   
4     100.0     100.0  ...     100.0     100.0     100.0     100.0     100.0   

   2022.1/4  2022.2/4  2022.3/4  2022.4/4  2023.1/4  
0    1974.9    2020.5    2019.6    2045.6    2020.7  
1     100.0     100.0     100.0     100.0     100.0  
2     100.0     100.0     100.0     100.0     100.0  
3    1370.4    1396.7    1398.9    1417.6    1416.2  
4     100.0     100.0     100.0     100.0     100.0  

[5 rows x 26 columns]

##### 4. 일자리의 질 지수 산출 데이터

### 2. DA

##### 1. 산업별 취업자

In [24]:
df_emp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 21 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   시도별      180 non-null    object
 1   산업별      180 non-null    object
 2   2023.01  180 non-null    int64 
 3   2023.02  180 non-null    int64 
 4   2023.03  180 non-null    int64 
 5   2023.04  180 non-null    int64 
 6   2023.05  180 non-null    int64 
 7   2023.06  180 non-null    int64 
 8   2023.07  180 non-null    int64 
 9   2023.08  180 non-null    int64 
 10  2023.09  180 non-null    int64 
 11  2013     180 non-null    object
 12  2014     180 non-null    object
 13  2015     180 non-null    object
 14  2016     180 non-null    object
 15  2017     180 non-null    int64 
 16  2018     180 non-null    int64 
 17  2019     180 non-null    int64 
 18  2020     180 non-null    int64 
 19  2021     180 non-null    int64 
 20  2022     180 non-null    int64 
dtypes: int64(15), object(6)
memory usage: 2

In [25]:
df_emp_age.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8316 entries, 0 to 8315
Data columns (total 25 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   성별         8316 non-null   object 
 1   행정구역별      8316 non-null   object 
 2   연령별        8316 non-null   object 
 3   항목         8316 non-null   object 
 4   단위         0 non-null      float64
 5   2013. 1/2  7854 non-null   float64
 6   2013. 2/2  7854 non-null   float64
 7   2014. 1/2  7854 non-null   float64
 8   2014. 2/2  7854 non-null   float64
 9   2015. 1/2  7854 non-null   float64
 10  2015. 2/2  7854 non-null   float64
 11  2016. 1/2  7854 non-null   float64
 12  2016. 2/2  7854 non-null   float64
 13  2017. 1/2  8316 non-null   float64
 14  2017. 2/2  7931 non-null   float64
 15  2018. 1/2  8316 non-null   float64
 16  2018. 2/2  8316 non-null   float64
 17  2019. 1/2  8316 non-null   float64
 18  2019. 2/2  8316 non-null   float64
 19  2020. 1/2  8316 non-null   float64
 20  2020. 2/